src/shogun/mathematics/linalg/GPUBackend.cpp

```diff
@@ -60,9 +61,23 @@ T GPUBackend::dot(const GPUVector<T> &a, const GPUVector<T> &b) const
 #endif
 }
 
+template <typename T>
+T GPUBackend::sum(const GPUVector<T> &vec) const
+{
+#ifdef HAVE_VIENNACL
+	return viennacl::linalg::sum(vec.gpuarray->GPUvec());
+#else
+	SG_SERROR("User did not register GPU backend. \n");
```

Comment:
why do we generate at all `GPUBackend` class if there's no gpu backend? i'm sorry but i'm very very confused what's happening here...

you should implement a `GPUBackend` using for example viannacl IF it's available. if not you just don't create one. like many other models, that if LAPACK was not present, the implementation was not built.